unity-renderer/Assets/Builder/Scripts/Gizmos/DCLBuilderTranslateGizmo.cs

```diff
@@ -8,11 +8,28 @@ public class DCLBuilderTranslateGizmo : DCLBuilderGizmo
 
         public override float TransformEntity(Transform entityTransform, DCLBuilderGizmoAxis axis, float axisValue)
         {
+            Vector3 initialEntityPosition = entityTransform.position;
+            if (snapFactor > 0)
+                initialEntityPosition = GetPositionRoundedToSnapFactor(entityTransform.position, axisValue);
+
             Vector3 move = activeAxis.transform.forward * axisValue;
-            Vector3 position = entityTransform.position + move;
+            Vector3 position = initialEntityPosition + move;
+
             entityTransform.position = position;
 
             return axisValue;
```

Comment:
I have concerns about these changes because this code is shared with the builder dapp

Reply:
I'll approve this. Let's revisit it if there is a issue next time builder needs an update